api/src/neo4j/shared/formatter.js

```diff
@@ -1,7 +1,25 @@
-const reformatExternalDbs = (externalDbs) => externalDbs.reduce((dbs, db) => {
+export const reformatExternalDbs = (externalDbs) => externalDbs.reduce((dbs, db) => {
   let dbRefs = dbs[db.dbName] || [];
   dbRefs = [...dbRefs, { id: db.externalId, url: db.url }];
   return { ...dbs, [db.dbName]: dbRefs };
 }, {});
 
-export default reformatExternalDbs;
+export const reformatCompartmentSVGs = (reaction) => [...reaction.compartments].sort((a, b) => a.id.localeCompare(b.id)).map((c) => {
+  const compartmentWithSVGs = reaction.compartmentSVGs.find(({ compartmentId }) => compartmentId === c.id);
+  const svgMaps = compartmentWithSVGs ? compartmentWithSVGs.compartmentSVGs.sort((a, b) => a.id.localeCompare(b.id)) : [];
+  return {
+    id: c.id,
+    customName: c.name,
+    svgMaps,
+  };
+})
+
+export const reformatSubsystemSVGs = (reaction) => [...reaction.subsystems].sort((a, b) => a.id.localeCompare(b.id)).map((s) => {
+  const subsystemsWithSVGs = reaction.subsystemSVGs.find(({ subsystemId }) => subsystemId === s.id);
+  const svgMaps = subsystemsWithSVGs ? subsystemsWithSVGs.subsystemSVGs.sort((a, b) => a.id.localeCompare(b.id)) : [];
+  return {
+    id: s.id,
+    customName: s.name,
+    svgMaps,
+  };
+});
```

Comment:
Nice with having this in the backend @inghylt. Do you want to try further refactoring these two functions? Maybe they can both call a new function called `reformatSVGs(component, svgs)`.